File: beeg.js
const axios = require('axios');
const cheerio = require('cheerio');

module.exports = async function beeg(query) {
  const url = `https://beeg.com/search?q=${encodeURIComponent(query)}`;
  const results = [];

  try {
    const { data } = await axios.get(url);
    const $ = cheerio.load(data);

    $('article').each((i, el) => {
      const title = $(el).find('.video-title').text().trim();
      const href = $(el).find('a').attr('href');

      if (title && href) {
        results.push({
          title,
          url: 'https://beeg.com' + href,
          duration: null,
          source: "Beeg"
        });
      }
    });
  } catch (err) {
    console.error("beeg error:", err.message);
  }

  return results;
};

File: scrapers/beeg.py
import requests
from bs4 import BeautifulSoup

def scrape_beeg(query, mode="straight", page=1):
    results = []
    # Beeg doesn't really have gay/trans filter, bias query
    if mode == "gay":
        query = f"gay {query}"
    elif mode == "trans":
        query = f"trans {query}"
    url = f"https://beeg.com/search?q={query}&page={page}"
    headers = {'User-Agent': 'Mozilla/5.0'}
    r = requests.get(url, headers=headers, timeout=10)
    soup = BeautifulSoup(r.content, "html.parser")
    for vid in soup.select("div.video-item"):
        try:
            a = vid.select_one("a")
            if not a or not a.has_attr("href"):
                continue
            title = a.get('title') or a.text.strip()
            video_url = "https://beeg.com" + a['href']
            img = vid.select_one("img")
            preview = img.get("src") if img else ""
            results.append({
                "title": title,
                "url": video_url,
                "preview": preview,
                "source": f"Beeg ({mode})"
            })
        except Exception:
            continue
    return results
